infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -370,65 +370,66 @@ public static void configureBootstrapOptions(@Nonnull ServerContext context,
-                        .collect(Collectors.toList());
-                enabledTlsCipherSuites = ciphers.toArray(new String[ciphers.size()]);
-            } else {
-                enabledTlsCipherSuites = new String[]{};
-            }
-
-            // Get the TLS protocols to enable
-            String protos = context.getServerConfig(String.class, "--tls-protocols");
-            if (protos != null) {
-                List<String> protocols = Pattern.compile(",")
-                        .splitAsStream(protos)
-                        .map(String::trim)
-                        .collect(Collectors.toList());
-                enabledTlsProtocols = protocols.toArray(new String[protocols.size()]);
-            } else {
-                enabledTlsProtocols = new String[]{};
-            }
-
-
-            try {
-                sslContext = SslContextConstructor.constructSslContext(true,
-                    context.getServerConfig(String.class, "--keystore"),
-                    context.getServerConfig(String.class, "--keystore-password-file"),
-                    context.getServerConfig(String.class, "--truststore"),
-                    context.getServerConfig(String.class,
-                        "--truststore-password-file"));
-            } catch (SSLException e) {
-                log.error("Could not build the SSL context", e);
-                throw new UnrecoverableCorfuError("Couldn't build the SSL context", e);
-            }
-        } else {
-            enabledTlsCipherSuites = new String[]{};
-            enabledTlsProtocols = new String[]{};
-            sslContext = null;
-        }
-
-        Boolean saslPlainTextAuth = context.getServerConfig(Boolean.class,
-                "--enable-sasl-plain-text-auth");
 
         // Generate the initializer.
         return new ChannelInitializer() {
             @Override
             protected void initChannel(@Nonnull Channel ch) throws Exception {
+
+                // Security variables
+                final SslContext sslContext;
+                final String[] enabledTlsProtocols;
+                final String[] enabledTlsCipherSuites;
+
+                // Security Initialization
+                Boolean tlsEnabled = context.getServerConfig(Boolean.class, "--enable-tls");
+                Boolean tlsMutualAuthEnabled = context.getServerConfig(Boolean.class,
+                        "--enable-tls-mutual-auth");
+                if (tlsEnabled) {
+                    // Get the TLS cipher suites to enable
+                    String ciphs = context.getServerConfig(String.class, "--tls-ciphers");
+                    if (ciphs != null) {
+                        List<String> ciphers = Pattern.compile(",")
+                                .splitAsStream(ciphs)
+                                .map(String::trim)
+                                .collect(Collectors.toList());
+                        enabledTlsCipherSuites = ciphers.toArray(new String[ciphers.size()]);
```

Comment:
rather than converting to List and then to array, it is possible to directly convert the stream to String[]
(e.g on a stream we can call .toArray(size -> new String[size])